epuck_ros2_cpp/src/driver.cpp

```diff
@@ -253,27 +265,27 @@ class EPuckPublisher : public rclcpp::Node {
     msg.range_min = 0.005 + SENSOR_DIST_FROM_CENTER;
     msg.range_max = 0.05 + SENSOR_DIST_FROM_CENTER;
     msg.ranges = std::vector<float>{
-      dist[3],       // -150
-      OUT_OF_RANGE,  // -135
-      OUT_OF_RANGE,  // -120
-      OUT_OF_RANGE,  // -105
-      dist[2],       // -90
-      OUT_OF_RANGE,  // -75
-      OUT_OF_RANGE,  // -60
-      dist[1],       // -45
-      OUT_OF_RANGE,  // -30
-      dist[0],       // -15
-      OUT_OF_RANGE,  // 0
-      dist[7],       // 15
-      OUT_OF_RANGE,  // 30
-      dist[6],       // 45
-      OUT_OF_RANGE,  // 60
-      OUT_OF_RANGE,  // 75
-      dist[5],       // 90
-      OUT_OF_RANGE,  // 105
-      OUT_OF_RANGE,  // 120
-      OUT_OF_RANGE,  // 135
-      dist[4],       // 150
+      dist[3] + SENSOR_DIST_FROM_CENTER,  // -150
+      OUT_OF_RANGE,                       // -135
+      OUT_OF_RANGE,                       // -120
+      OUT_OF_RANGE,                       // -105
+      dist[2] + SENSOR_DIST_FROM_CENTER,  // -90
+      OUT_OF_RANGE,                       // -75
+      OUT_OF_RANGE,                       // -60
+      dist[1] + SENSOR_DIST_FROM_CENTER,  // -45
+      OUT_OF_RANGE,                       // -30
+      dist[0] + SENSOR_DIST_FROM_CENTER,  // -15
+      distTof + SENSOR_DIST_FROM_CENTER,  // 0
+      dist[7] + SENSOR_DIST_FROM_CENTER,  // 15
+      OUT_OF_RANGE,                       // 30
+      dist[6] + SENSOR_DIST_FROM_CENTER,  // 45
+      OUT_OF_RANGE,                       // 60
+      OUT_OF_RANGE,                       // 75
+      dist[5] + SENSOR_DIST_FROM_CENTER,  // 90
+      OUT_OF_RANGE,                       // 105
+      OUT_OF_RANGE,                       // 120
+      OUT_OF_RANGE,                       // 135
+      dist[4] + SENSOR_DIST_FROM_CENTER,  // 150
```

Comment:
Is this a common practice in ROS to compute the distance from the center of the robot?
To me this looks duplicated with the fact that the position of the sensors is set:
https://github.com/cyberbotics/epuck_ros2/pull/7/files#diff-fac28970bca9d5d66bd5d9c3c778bba7R151

Reply:
The one you pointed out is a transform of the sensor itself (used for Range topic). LaserScan topic is made of "virtual LiDAR" (frame `laser_scanner`). We need LaserScan because a lot of ROS nodes use it for mapping.

Basically, we have Range topic for we do everything regularly, and we have LaserScan topic for which we improvise (as we don't have LiDAR).

Reply:
Oh yes you are completely right, I mixed both topics, it then perfectly makes sense!